tests/testthat/resources/test-custom-root-dir.Rmd

```diff
@@ -0,0 +1,47 @@
+---
+title: "reticulate"
+---
+
+Change the knit root directory to the temporary R session.
+See https://github.com/rstudio/reticulate/issues/1526 for context.
+
+```{r setup}
+knitr::opts_knit$set("root.dir" = tempdir())
+```
+
+```{r confirm-wd}
+getwd()
+```
+
+```{r configure-python}
+library("reticulate")
+matplotlib <- import("matplotlib")
+```
+
+```{r}
+reticulate::py_config()
+```
+
+The R plot is saved relative to the location of the Rmd file:
+
+```{r r-plot}
+plot(1:10)
+```
+
+The Python plot is also saved relative to the location of the Rmd file (not root.dir):
+
+```{python py-plot}
+import matplotlib.pyplot as plt
+
+x = range(1, 10)
+plt.plot(x, x)
+```
+
+```{r}
+knitr::opts_knit$get("root.dir")
+
+if( file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "r-plot-1.png")))    stop("figure saved in wrong place")
+if( file.exists(file.path(knitr::opts_knit$get("root.dir"), "figure", "py-plot-1.png")))   stop("figure saved in wrong place")
+if(!file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "r-plot-1.png")))  stop("figure saved in wrong place")
+if(!file.exists(file.path(knitr::opts_knit$get("output.dir"), "figure", "py-plot-1.png"))) stop("figure saved in wrong place")
```

Comment:
Just a note that `figure` folder is only valid for `knitr::knit()` of this document; I see you are using `rmarkdown::render()` in tests, for the intermediate dir will be different and not `figure`. It will be of the form `*_files/figure_<format>`

So with this test files it will be something like `test-custom-root-dir_files/figure-html`

* If this folders and png files are in `root.dir` this is wrong place, 
* If they are in `output.dir` this is correct place. 

Sorry for my example being misleading - I gave one that did not involved **rmarkdown** and only **knitr**.

Reply:
Thanks @cderv! I *just* figured out why tests were failing and then noticed your comment :)